AzureMonitorAgent/agent.py

```diff
@@ -203,25 +204,41 @@ def install():
                                          retry_check = retry_if_dpkg_locked,
                                          final_check = final_check_if_dpkg_locked)
 
+    default_configs = {
+        "ENABLE_MCS" : "true",
+        "MCS_ENDPOINT" : "eastus.amcs.ppe.control.monitor.azure.com",
+        "AZURE_ENDPOINT" : "https://management.azure.com/",
+        "ADD_REGION_TO_MCS_ENDPOINT" : "false",
+        "OMS_TLD" : "int2.microsoftatlanta-int.com",
+        "customResourceId" : "/subscriptions/42e7aed6-f510-46a2-8597-a5fe2e15478b/resourcegroups/amcs-test/providers/Microsoft.OperationalInsights/workspaces/amcs-pretend-linuxVM"
+    }
+    config_file = "/etc/default/mdsd"
+    config_updated = False
     try:
-        if os.path.isfile("/etc/default/mdsd"):
-            with open("/etc/default/mdsd", "a+") as f:
+        if os.path.isfile(config_file):
+            data = ""
+            with open(config_file, "r") as f:
                 data = f.read()
-                if "ENABLE_MCS=true" not in data:
-                    f.write("\n")
-                    f.write("export ENABLE_MCS=true\n")
-                    
-                # for integration testing
-                if "MCS_ENDPOINT=mcs.azure.com" not in data:
-                    f.write("export MCS_ENDPOINT=eastus.amcs.ppe.control.monitor.azure.com\n")
-                if "AZURE_ENDPOINT=https://management.azure.com/" not in data:
-                    f.write("export AZURE_ENDPOINT=https://management.azure.com/\n")
-                if "ADD_REGION_TO_MCS_ENDPOINT" not in data:
-                    f.write("export ADD_REGION_TO_MCS_ENDPOINT=false\n")
-                if "OMS_TLD" not in data:
-                    f.write("export OMS_TLD=int2.microsoftatlanta-int.com\n")
-                if "customResourceId" not in data:
-                    f.write("export customResourceId=/subscriptions/42e7aed6-f510-46a2-8597-a5fe2e15478b/resourcegroups/amcs-test/providers/Microsoft.OperationalInsights/workspaces/amcs-pretend-linuxVM\n")
+                for var in default_configs.keys():
+                    pattern = "(#*)export " + str(var) + "=.*$"
+                    r_obj = re.compile(pattern, flags=re.MULTILINE)
```

Comment:
how did you verify this?

Reply:
I copied over that file to a vm in canary that had ama installed and then used that agent.py to uninstall/install/update. While changing the mdsd file and checking if it was updated correctly